### ISSUE TITLE
model.py

### ISSUE THREAD

**Issue by laithdabbas** [laithdabbas/chest_#1]
import yfinance as yf
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.preprocessing import MinMaxScaler
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import LSTM, Dense
import pickle

# -----------------------------
# Logistic Regression
# -----------------------------
# Download stock data (example: AAPL)
data = yf.download("AAPL", start="2018-01-01", end="2024-01-01")
data['Target'] = (data['Close'].shift(-1) > data['Close']).astype(int)
data = data.dropna()

# Features
X = data[['Close', 'Volume']].shift(1).dropna()
y = data['Target'][1:]  # align target

# Train-test split
X_train, X_test, y_train, y_test = train_test_split(X, y, shuffle=False, test_size=0.2)

# Train Logistic Regression
model = LogisticRegression()
model.fit(X_train, y_train)

# Save Logistic Regression model
with open("stock_model.pkl", "wb") as f:
    pickle.dump(model, f)

print("✅ Logistic Regression model trained and saved as stock_model.pkl")

# -----------------------------
# LSTM Model (Optional)
# -----------------------------
scaler = MinMaxScaler()
scaled = scaler.fit_transform(data[['Close']])

def create_sequences(data, window=30):
    X_seq, y_seq = [], []
    for i in range(len(data)-window-1):
        X_seq.append(data[i:i+window])
        y_seq.append(1 if data[i+window] > data[i+window-1] else 0)
    return np.array(X_seq), np.array(y_seq)

X_seq, y_seq = create_sequences(scaled, window=30)
X_train_seq, X_test_seq = X_seq[:-200], X_seq[-200:]
y_train_seq, y_test_seq = y_seq[:-200], y_seq[-200:]

model_lstm = Sequential([
    LSTM(50, return_sequences=False, input_shape=(30,1)),
    Dense(1, activation="sigmoid")
])
model_lstm.compile(optimizer="adam", loss="binary_crossentropy", metrics=["accuracy"])
model_lstm.fit(X_train_seq, y_train_seq, epochs=5, batch_size=16, validation_split=0.1)

loss, acc = model_lstm.evaluate(X_test_seq, y_test_seq)
print("LSTM Accuracy:", acc)
